android/app/src/main/java/org/fossasia/openevent/core/feedback/FeedbackFragment.java

```diff
@@ -0,0 +1,155 @@
+        feedbackFragmentViewModel = ViewModelProviders.of(this).get(FeedbackFragmentViewModel.class);
+        if (AuthUtil.isUserLoggedIn()) {
+            if (NetworkUtils.haveNetworkConnection(getContext())) {
+                swipeRefreshLayout.setRefreshing(true);
+                downloadFeedbacks();
+            }
+            loadData();
+        } else {
+            redirectToLogin();
+        }
+        return view;
+    }
+
+    private void redirectToLogin() {
+        Toast.makeText(getContext(), "User need to be logged in!", Toast.LENGTH_SHORT).show();
+        Intent intent = new Intent(getActivity(), LoginActivity.class);
+        startActivity(intent);
+    }
+
+    private void downloadFeedbacks() {
+        feedbackFragmentViewModel.downloadFeedbacks().observe(this,this::onFeedbacksDownloadDone);
+    }
+
+    private void setUpRecyclerView() {
+        feedbacksRecyclerView.setVisibility(View.VISIBLE);
+        DividerItemDecoration itemDecor = new DividerItemDecoration(getActivity(), LinearLayoutManager.VERTICAL);
+        feedbacksRecyclerView.addItemDecoration(itemDecor);
+        feedbacksListAdapter = new FeedbacksListAdapter(feedbackList);
+        feedbacksRecyclerView.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
+        feedbacksRecyclerView.setNestedScrollingEnabled(false);
+        feedbacksRecyclerView.setAdapter(feedbacksListAdapter);
+        feedbacksRecyclerView.setItemAnimator(new DefaultItemAnimator());
+    }
+
+    private void handleVisibility() {
+        if (!feedbackList.isEmpty()) {
+            feedbackHeaderView.setVisibility(View.VISIBLE);
+            feedbacksRecyclerView.setVisibility(View.VISIBLE);
+            noFeedbacksView.setVisibility(View.GONE);
+        } else {
+            feedbackHeaderView.setVisibility(View.GONE);
+            feedbacksRecyclerView.setVisibility(View.GONE);
+        }
+    }
+
+    private void loadData() {
+        feedbackFragmentViewModel.getFeedback().observe(this, feedbacks -> {
+            feedbackList.clear();
+            feedbackList.addAll(feedbacks);
+            feedbacksListAdapter.notifyDataSetChanged();
+            handleVisibility();
+        });
+    }
+
+    public void onFeedbacksDownloadDone(boolean status) {
+        if (!status) {
+            Timber.d("Feedbacks Download failed");
+            if (getActivity() != null && swipeRefreshLayout != null) {
+                Snackbar.make(swipeRefreshLayout, getActivity().getString(R.string.refresh_failed), Snackbar.LENGTH_LONG)
+                        .setAction(R.string.retry_download, view -> refresh()).show();
+            }
+        }
+        if (swipeRefreshLayout != null)
+            swipeRefreshLayout.setRefreshing(false);
+    }
+
+    private void refresh() {
+        NetworkUtils.checkConnection(new WeakReference<>(getContext()), new NetworkUtils.NetworkStateReceiverListener() {
```

Comment:
Use direct check, not callback

Reply:
@iamareebjamal callback is used in many other VM's also should I create an issue and update it to direct check?